src/OpenTelemetry.Exporter.Prometheus.AspNetCore/README.md

```diff
@@ -0,0 +1,106 @@
+# Prometheus Exporter AspNetCore for OpenTelemetry .NET
+
+## Prerequisite
+
+* [Get Prometheus](https://prometheus.io/docs/introduction/first_steps/)
+
+## Steps to enable OpenTelemetry.Exporter.Prometheus.AspNetCore
+
+### Step 1: Install Package
+
+Install
+
+```shell
+dotnet add package OpenTelemetry.Exporter.Prometheus.AspNetCore
+```
+
+### Step 2: Configure OpenTelemetry MeterProvider
+
+* When using OpenTelemetry.Extensions.Hosting package on .NET Core 3.1+:
+
+    ```csharp
+    services.AddOpenTelemetryMetrics(builder =>
+    {
+        builder.AddPrometheusExporter();
+    });
+    ```
+
+* Or configure directly:
+
+    Call the `AddPrometheusExporter` `MeterProviderBuilder` extension to
+    register the Prometheus exporter.
+
+    ```csharp
+    using var meterProvider = Sdk.CreateMeterProviderBuilder()
+        .AddPrometheusExporter()
+        .Build();
+    ```
+
+### Step 3: Configure Prometheus Scraping Endpoint
+
+* On .NET Core 3.1+ register Prometheus scraping middleware using the
+  `UseOpenTelemetryPrometheusScrapingEndpoint` extension:
+
+    ```csharp
+    public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+    {
+        app.UseOpenTelemetryPrometheusScrapingEndpoint();
+        app.UseRouting();
+        app.UseEndpoints(endpoints =>
+        {
+            endpoints.MapControllers();
+        });
+    }
+    ```
+
+    Overloads of the `UseOpenTelemetryPrometheusScrapingEndpoint` extension are
+    provided to change the path or for more advanced configuration a predicate
+    function can be used:
+
+    ```csharp
+    public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+    {
+        app.UseOpenTelemetryPrometheusScrapingEndpoint(
+            context => context.Request.Path == "/internal/metrics"
+                && context.Connection.LocalPort == 5067);
+        app.UseRouting();
+        app.UseEndpoints(endpoints =>
+        {
+            endpoints.MapControllers();
+        });
+    }
+    ```
+
+## Configuration
+
+The `PrometheusExporter` can be configured using the `PrometheusExporterOptions`
+properties. Refer to
+[`TestPrometheusExporter.cs`](../../examples/Console/TestPrometheusExporter.cs)
+for example use.
+
+### ScrapeEndpointPath
+
+Defines the path for the Prometheus scrape endpoint for
+either the HTTP listener or the middleware registered by
```

Comment:
the comment seems outdated, as it is referring to Http Listener?

Reply:
Resolved in https://github.com/open-telemetry/opentelemetry-dotnet/pull/3430.